python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+try:
+    import lsst.eotest.sensor as sensorTest
+except ImportError:
+    raise RuntimeError("Error importing eotest")
+
+import os
+import glob
+
+
+class CpTaskConfig(pexConfig.Config):
+    """Config class for the calibration products production (CP) task."""
+
+    fe55 = pexConfig.ConfigurableField(
+        target=sensorTest.Fe55Task,
+        doc="The eotest Fe55 analysis task. Should not be retragetted.",
+    )
+    doFe55 = pexConfig.Field(
+        dtype=bool,
+        doc="Run the Fe55 task to measure gains?",
+        default=True,
+    )
+    readNoise = pexConfig.ConfigurableField(
+        target=sensorTest.ReadNoiseTask,
+        doc="The eotest read noise task. Should not be retragetted.",
+    )
+    doReadNoise = pexConfig.Field(
+        dtype=bool,
+        doc="Run the readNoiseTask to measure read noise?",
+        default=True,
+    )
+    brightPixels = pexConfig.ConfigurableField(
+        target=sensorTest.BrightPixelsTask,
+        doc="The eotest bright pixel/column finding task. Should not be retragetted.",
+    )
+    doBrightPixels = pexConfig.Field(
+        dtype=bool,
+        doc="Run the brightPixelTask to find the bright pixels?",
+        default=True,
+    )
+    darkPixels = pexConfig.ConfigurableField(
+        target=sensorTest.DarkPixelsTask,
+        doc="The eotest dark pixel/column finding task. Should not be retragetted.",
+    )
+    doDarkPixels = pexConfig.Field(
+        dtype=bool,
+        doc="Run the darkPixelsTask to find the dark pixels?",
+        default=True,
+    )
+    traps = pexConfig.ConfigurableField(
+        target=sensorTest.TrapTask,
+        doc="The eotest trap-finding task. Should not be retragetted.",
+    )
+    doTraps = pexConfig.Field(
+        dtype=bool,
+        doc="Run the trapTask to find the traps?",
+        default=True,
+    )
+    cte = pexConfig.ConfigurableField(
+        target=sensorTest.CteTask,
+        doc="The eotest CTE analysis task. Should not be retragetted.",
+    )
+    doCTE = pexConfig.Field(
+        dtype=bool,
+        doc="Run the CTE task to measure the CTE?",
+        default=True,
+    )
+    ptc = pexConfig.ConfigurableField(
+        target=sensorTest.PtcTask,
+        doc="The eotest PTC analysis task. Should not be retragetted.",
+    )
+    doPTC = pexConfig.Field(
+        dtype=bool,
+        doc="Run the PTC task to measure the photon transfer curve?",
+        default=True,
+    )
+    flatPair = pexConfig.ConfigurableField(
+        target=sensorTest.FlatPairTask,
+        doc="The eotest flat-pair analysis task. Should not be retragetted.",
+    )
+    doFlatPair = pexConfig.Field(
+        dtype=bool,
+        doc="Run the flatPair task?",
+        default=True,
+    )
+    eotestOutputPath = pexConfig.Field(
+        dtype=str,
+        doc="Path to which to write the eotest output results. Madatory runtime arg for running eotest.",
+        default='',
+    )
```

Comment:
I had a way of generating this at run-time from the butler, but it was very hacky, and more importantly, because it needs to be set in all the subTasks at instantiation-time rather than run-time there really isn't a way around this as they're frozen.

I agree it's not good like this, but this won't be run from the command line, and nobody will suffer needlessly from this because running eotest is something that won't be done much at all.